app/src/swap/swap_utils.c

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ *   (c) 2016 Ledger
+ *   (c) 2018 - 2023 Zondax AG
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#include "bignum.h"
+#include "crypto.h"
+#include "lib_standard_app/swap_lib_calls.h"
+#include "swap.h"
+#include "zxformat.h"
+
+#define DOT_COIN_AMOUNT_DECIMAL_PLACES 10
+#define DOT_COIN_TICKER                "DOT "
+////////////////////////////////////////////////////////////////
+
+zxerr_t bytesAmountToStringBalance(uint8_t *amount, uint8_t amount_len, char *out, uint8_t out_len) {
```

Comment:
Should we check for NULL pointers here?

Reply:
why not, it should never be NULL by design but that's not a lot to add